sdk/storage/Azure.Storage.Files/src/Models/FileSmbProperties.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Azure.Storage.Files.Models
+{
+    /// <summary>
+    /// The SMB properties for a file.
+    /// </summary>
+    public struct FileSmbProperties : IEquatable<FileSmbProperties>
+    {
+        /// <summary>
+        /// The file system attributes for this file.
+        /// </summary>
+        public NtfsAttributes? FileAttributes { get; set; }
+
+        /// <summary>
+        /// The file permission key of the file's FilePermission.
+        /// </summary>
+        public string FilePermissionKey{ get; set; }
```

Comment:
Why isn't there a `public string FilePermission { get; set; }`?

Reply:
CreateFile, GetFileProperties, GetFile, and SetFileProperties always return a FilePermissionKey header in the response, and never return a FilePermission header. Even if you include a FilePermission in CreateFile or SetFileProperties, the response will contain a FilePermissionKey.

Reply:
Sure, but we also take this as an argument to all of the methods that have SMB properties.  I think we should include the FilePermission here instead of passing it as a separate argument.

Reply:
The only methods that accept SMB properties are CreateFile, SetFileProperties, CreateDirectory, and SetDirectoryProperties.

It is a strange experience that you can set a FilePermission and you get a FilePermissionId back, but this is backend business logic we can't change.  I think it is more intuitive for the user to keep it FilePermission separate, so they aren't confused when SmbProperties.FilePermission is always null.

The way you get a FilePermission is to call Share.GetFilePermission(filePermissionId).  FilePermissions are almost their own first-class object.

Reply:
Let's leave it as you have it for now and open a work item to start a thread with the Arch Board about the best way to handle this.  I think it makes more sense in `FileSmbProperties`, but it's obviously not cut and dry.